jsonrpc/filter_manager.go

```diff
@@ -486,6 +489,17 @@ func (f *FilterManager) appendLogsToFilters(header *types.Header, removed bool)
 	}
 
 	for indx, receipt := range receipts {
+		if receipt.TxHash == types.ZeroHash {
+			// Get a block where tx was sealed
+			block, ok := f.store.GetBlockByHash(header.Hash, true)
+			if !ok {
+				f.logger.Error("could not find block in store", "hash", header.Hash.String())
+
+				return nil
+			}
+			// Extract tx Hash
+			receipt.TxHash = block.Transactions[indx].Hash
+		}
```

Comment:
I don't think it's efficient code because GetBlockByHash will be called several times if multiple receipts don't have TxHash. You might want to get block in the outer side of loop.

Reply:
This is only for backwards compatibility. No new tx will have receipt.TxHash == types.ZeroHash. 
What do you think is more efficient, leaving like this and fetching only for old logs, or fetching block every time once?

Reply:
ping @Kourin1996

Reply:
I understood it's for backwards compatibility. But it doesn't need to fetch block multiple times, does it? Why don't you save fetched block to variable?

Reply:
That's a good idea will do that